accord-core/src/main/java/accord/primitives/Writes.java

```diff
@@ -18,20 +18,21 @@
 
 package accord.primitives;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
 import accord.api.Write;
 import accord.local.SafeCommandStore;
 import accord.utils.async.AsyncChain;
 import accord.utils.async.AsyncChains;
 
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Objects;
-
 public class Writes
 {
     public static final AsyncChain<Void> SUCCESS = AsyncChains.success(null);
     public final TxnId txnId;
     public final Timestamp executeAt;
+    // TODO (review): Don't these have to be keys so why Seekables?
```

Comment:
Can we write a range? I think we can only write keys. By the time transaction execution progresses this far that collection should only contain keys I think.

It does look like there is a single path where we could construct `Writes` with ranges and that is `Txn.execute` which asks the `Update` for the keys, and I do sort of question whether the `Update` interface there should be able to return a range because `Write.apply` is only expecting keys (and `TxnWrite` hard codes a cast), and the implementations of `Update` also concretely use `Keys`.

It's definitely possible that `Write.apply` could be provided a range and should then apply all keys intersecting that range, but nothing supports/implements that yet.

Reply:
no I guess not, all range "writes" are updating accord metadata like eviction thresholds etc